Fixing Issue no. 45353 , highlighting the active module in the sidebar

DEVANSH-GAJJAR opened the pull request:
<!--

Some key notes before you open a PR:

 1. Select which branch should this PR be merged in?
 2. PR name follows [convention](http://karma-runner.github.io/4.0/dev/git-commit-msg.html)
 3. All tests pass locally, UI and Unit tests
 4. All business logic and validations must be on the server-side
 5. Update necessary Documentation
 6. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes


Also, if you're new here

- Documentation Guidelines => https://github.com/frappe/erpnext/wiki/Updating-Documentation

- Contribution Guide => https://github.com/frappe/frappe/blob/develop/.github/CONTRIBUTING.md

- Pull Request Checklist => https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

-->



> Explain the **details** for making this change. What existing problem does the pull request solve?

Right now, when we switch between workspaces, the sidebar doesn’t show which one is open, so it’s a bit confusing. I added a small code change in workspace.js so that the current workspace gets an .active highlight in the sidebar. This makes it easier to know where you are while using ERPNext.

<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->

> Screenshots/GIFs
The below diagram shows the before/after effect of the change: the currently open workspace is now visibly highlighted in the sidebar

<img width="1051" height="277" alt="image" src="https://github.com/user-attachments/assets/8d56fd77-d645-4cbb-8c9b-ddf1809a8c1a" />


<!-- Add images/recordings to better visualize the change: expected/current behavior -->

DEVANSH-GAJJAR committed:
Fixing Issue no. 45353 , highlighting the active module in the sidebar

akhilnarang reviewed (changes requested):
Can you please fix the formatting (install and run [pre-commit](https://pre-commit.com))? 

And also fix the typos in the comment, and not use all capitals there.

DEVANSH-GAJJAR committed:
fix: highlight active sidebar item after navigation

DEVANSH-GAJJAR commented:
i have updated the PR , with the required change . Please review when you get a chance 

Updated the commit message to follow semantic commit conventions

thank you

iamejaaz commented:
Hello,
Could you please attach a screenshot or video showing how it looks in the UI?

DEVANSH-GAJJAR commented:
<img width="169" height="415" alt="11111" src="https://github.com/user-attachments/assets/a5bf9b34-6830-4166-9920-e4f312f19619" />


Hello Sir , 

This is the  Screenshot of highlighting the active sidebar (here the home is highlighted )
I hope everything is great 


thank you

iamejaaz commented:
Hello,

I can see that the sidebar menu items are already highlighted. Also, the current implementation using setTimeout doesn’t seem ideal.

Additionally, as shown in Frappeverse 2025, the sidebar and workspace structure is going to change soon.

Closing this PR for now. Let me know if I’m missing anything — happy to reopen if needed.

<img width="2690" height="1260" alt="image" src="https://github.com/user-attachments/assets/1e1add7c-fbee-4298-b5f4-e5321a30f432" />
